cpu/sam0_common/periph/uart.c

```diff
@@ -389,24 +420,25 @@ static inline void irq_handler_tx(unsigned uartnum)
 static inline void irq_handler(unsigned uartnum)
 {
     uint32_t status = dev(uartnum)->INTFLAG.reg;
+    dev(uartnum)->INTFLAG.reg = status;
```

Comment:
This part clears IRQ flags that aren't handled by this ISR. We should mask-out those flags.